### ISSUE TITLE
Is this accurate?

### ISSUE THREAD

**Issue by dpalme3160** [realm/SwiftLint#3119]
let tempArray: Array = (dict["Questions"]! as? [Dictionary<String, AnyObject>])!

is throwing the following error: 

Syntactic Sugar Violation: Shorthand syntactic sugar should be used, i.e. [String: Int] instead of Dictionary<String, Int>. (syntactic_sugar)

I tried changing it to [String, AnyObject] but that then caused all kinds of other errors, actual errors.  Not sure why this is saying its a violation.

**Comment by jpsim** [comment]
That rule is suggesting the following:

```swift

let tempArray: Array = (dict["Questions"]! as? [[String: AnyObject]])!
```

Although I’d generally encourage you to consider more typesafe ways to extract and model that data.
